### PR TITLE
Add terraform script for complete deployment of the Registry in the Dev Account

### LINKED ISSUE #472: Create an IAM policy for tenant data operator
### Are you sure this is not a new requirement or bug?

No

### Task Type

Sub-task

### 💡 Description

They will need to use the opensearch dashboards
.
Read from the full registry and write in their own node opensearch index.

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
## 🗒️ Summary

This PR initiates a terraform script which deployed opensearch and the needed policies to start.

A follow up action is going to be able to initialize the registry with registry-mgr and the reference dataset.

It is ready to be merged as it is used for deployment in dev with validated integration tests as a basis for the ISRO node creation.

## ⚙️ Test Data and/or Report

Integration test validated.

## ♻️ Related Issues

- fixes https://github.com/NASA-PDS/registry/issues/472

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -55,9 +55,9 @@ module "api_gateway" {
   common_tags          = var.common_tags
 }
 
+# TL: not ready for integration as we have a deadline for the ISRO node creation in dev, step done manuallyt
```
Comment (nutjob4life) [inline terraform/main.tf:59]: @tloubrieu-jpl I feel seen! 😁   (Which is also how some people pronounce my name 🤣)

Thank you for doing this 👍

**Comment by jordanpadams** [comment]
@tloubrieu-jpl can you review the comments from Copilot and either either dismiss the proposed updates or assign copilot to fix them?

File: terraform/run-init-on-aws.py [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+        """
+        self.print_section("🚀", f"Running {purpose}...")
+        print(f"   Image: {self.env_vars['REG_LOADER_IMAGE']}")
+        print(f"\n📋 Container logs:")
+        print("-" * 50)
+
+
+
+        docker_cmd = [
+            "docker", "run", "--rm", "-it",
```
[the remark this answers is not in the document]
Comment (tloubrieu-jpl) [reply terraform/run-init-on-aws.py:324]: The script is meant to run in interactive environment for now. We can remove that later.

File: terraform/lambda/src/get_awskeys_from_cognitojwt.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+        # Search for the kid in the downloaded JWSK
+        public_key = None
+        for key in jwsk['keys']:
+            if kid == key.get('kid', ''):
+                public_key = jwk.construct(key)
+                break
+
+        if public_key is None:
+            raise ValueError("No keys found in JWSK.")
```
[the remark this answers is not in the document]
Comment (tloubrieu-jpl) [reply terraform/lambda/src/get_awskeys_from_cognitojwt.py:160]: That is a legacy script, I don't want to take the risk of breaking it. The script should be obsolete too at some point when we migrate to managed opensearch.

File: terraform/run-init-on-aws.py [hunk 4]
```diff
@@ -0,0 +1,585 @@
...
+            self.env_vars.setdefault("TEST_DATA_URL",
+                "https://github.com/NASA-PDS/registry-ref-data/releases/download/Latest/custom-datasets.tar.gz")
+            self.env_vars.setdefault("TEST_DATA_LIDVID",
+                "urn:nasa:pds:mars2020.spice::1.0 urn:nasa:pds:mars2020.spice::2.0 urn:nasa:pds:mars2020.spice::3.0")
+            self.env_vars.setdefault("CONTAINER_HARVEST_DATA_DIR", "/data")
+            self.env_vars.setdefault("REG_DATA_VOLUME", "data-volume")
+
+    def add_locally_defined_env_vars(self) -> None:
+        for k , v in os.environ.items():
+            self.env_vars[k] = v
```
[the remark this answers is not in the document]
Comment (tloubrieu-jpl) [reply terraform/run-init-on-aws.py:217]: That is the intention. .env is pre-existing configuration that we want to override in some cases.

File: terraform/run-init-on-aws.py [hunk 5]
```diff
@@ -0,0 +1,585 @@
...
+
+        # Extract tar.gz archive
+        self.print_section("📦", "Extracting test data...")
+        extract_dir = output_dir / "extracted"
+        extract_dir.mkdir(parents=True, exist_ok=True)
+
+        try:
+            with tarfile.open(download_path, "r:gz") as tar:
+                tar.extractall(path=extract_dir)
+                print(f"   Extracted {len(tar.getmembers())} files to: {extract_dir}\n")
```
[the remark this answers is not in the document]
Comment (tloubrieu-jpl) [reply terraform/run-init-on-aws.py:425]: Let's see that when that becomes an issue.

File: terraform/run-init-on-aws.py [hunk 6]
```diff
@@ -0,0 +1,585 @@
...
+
+        self.credentials_endpoint = self.run_command(
+            [self.tf_cmd, "output", "-raw", "credentials_endpoint"],
+            cwd=self.tf_working_dir,
+        )
+
+        node_list_json = self.run_command(
+            [self.tf_cmd, "output", "-json", "node_list"],
+            cwd=self.tf_working_dir,
+        )
```
[the remark this answers is not in the document]
Comment (tloubrieu-jpl) [reply terraform/run-init-on-aws.py:126]: We don't worry about that yet
